clean-code-challanges/src/main/java/PigLatinTranslator.java

```diff
@@ -18,6 +22,112 @@
+  }
+
+}
+
+
+class Phrase {
+  private List<String> sourcePhrase;
+  private List<String> translatedPhrase = new ArrayList<>();
+
+  Phrase(String sourcePhrase){
+    this.sourcePhrase = Arrays.asList(sourcePhrase.split(" "));
+  }
+
+  public void translateToPigLatin() {
+    for (String sourceWord : sourcePhrase){
+      translatedPhrase.add(translateWordToPigLatin(sourceWord));
+    }
+  }
+
+  public String getTranslatedPhrase() {
+    return String.join(" ",translatedPhrase);
+  }
+
+
+  private String translateWordToPigLatin(String sourceWord){
+      Word word = new Word(sourceWord);
+      if  (word.beginsWithTripleConsonantCluster() || word.firstConsonantFollowedByQU()){
+        word.moveFirstThreeLettersToTheEndOfWord();
+
+      }else if (word.beginsWithDoubleConsonantCluster()) {
+        word.moveFirstTwoLettersToTheEndOfWord();
+
+      }else if (!word.beginsWithVowelSound() && !word.beginsWithVowelSoundCluster() ){
+        word.moveFirstLetterToTheEndOfWord();
+      }
+      word.appendSuffix();
+
+      return word.getString();
+  }
+
+}
+
+
+
+class Word {
+  private String word;
+  private ArrayList<String> vowelSounds = new ArrayList<>(Arrays.asList("a", "e", "i", "o", "u"));
+  private ArrayList<String> doubleConsonantClusters = new ArrayList<>(Arrays.asList("ch", "th", "qu"));
+  private ArrayList<String> tripleConsonantClusters = new ArrayList<>(Arrays.asList("thr", "sch"));
+  private ArrayList<String>  vowelSoundsCluster = new ArrayList<>(Arrays.asList("yt", "xr"));
+  private String specialCombo = "qu";
+  private String suffix = "ay";
+
+
+  Word(String word){
+    this.word = word;
+  }
+
+
+  public boolean beginsWithVowelSound(){
+    String firstLetter =  word.substring(0,1).toLowerCase();
+    return vowelSounds.contains(firstLetter);
+  }
+
+  public boolean beginsWithVowelSoundCluster(){
+    String firstTwoLetters =  word.substring(0,2).toLowerCase();
+    return vowelSoundsCluster.contains(firstTwoLetters);
+  }
+
+  public boolean beginsWithDoubleConsonantCluster(){
+    String firstTwoLetters =  word.substring(0,2).toLowerCase();
+    return doubleConsonantClusters.contains(firstTwoLetters);
+  }
+
+  public boolean beginsWithTripleConsonantCluster(){
+    String firstThreeLetters = word.substring(0,3).toLowerCase();
+    return tripleConsonantClusters.contains(firstThreeLetters);
+  }
+
+  public boolean firstConsonantFollowedByQU( ){
+    String secondAndThirdLetter =  word.substring(1,3).toLowerCase();
+    return (secondAndThirdLetter.equals(specialCombo));
   }
+
+  public void appendSuffix() {
+    word = word + suffix;
+  }
+
+  public void moveFirstLetterToTheEndOfWord(){
+    word = word.substring(1) + word.charAt(0);
+  }
+
+  public void moveFirstTwoLettersToTheEndOfWord(){
+    word = word.substring(2) + word.charAt(0) + word.charAt(1);
+  }
+
+  public void moveFirstThreeLettersToTheEndOfWord(){
+    word = word.substring(3) + word.charAt(0) + word.charAt(1) + word.charAt(2);
+  }
```

Comment:
sehr cleaner code und verständlich und leserlich. nur fallen mir code-wiederholungen auf, könnte man zB die move- und beginsWith-funktionen  in wenigere funktionen ausarbeiten? sonst wie oben commentiert, es werden nicht alle fälle abgedeckt.

Reply:
Danke für dein Feedback, habe die Methoden ein weinig gekürzt.